### PR TITLE
fix: show js function execution errors in debugger

### LINKED ISSUE #14503: [Bug]: JS object is in an error state but no error is displayed on debugger or browser console
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

JS object goes into error during execution but no error is displayed either on errors tab on the debugger or on browser console.

### Steps To Reproduce

1.  Go to JS object at [this](https://release.app.appsmith.com/app/scratch-pad/js-error-issue-62a7690dd901b06688ea14fb/edit/jsObjects/62a7690dd901b06688ea1513) location and run function `myFun1`
2. Observe the execution error 

### Public Sample App

https://release.app.appsmith.com/app/scratch-pad/js-error-issue-62a7690dd901b06688ea14fb/edit/jsObjects/62a7690dd901b06688ea1513

### Version

Cloud

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

This PR introduces js function execution errors in the debugger. These errors behave in the same way as those of Apis

They get removed when

- [x] The function gets deleted

- [x] Cause of error is fixed and function is re-run

- [x] JS Object is deleted

Fixes #14503 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Cypress tests

### Test Plan
- [x] [[Test Case]: verify, errors appears in debugger of JS editor when a JS object is in error state](https://github.com/appsmithorg/TestSmith/issues/1895)
- [ ] [[Test Case]: verify, errors appears in browser console when a JS object is in error state](https://github.com/appsmithorg/TestSmith/issues/1896) (Manual)
- [ ] [[Test Case]: verify, errors appears in debugger pane of JS editor and browser console](https://github.com/appsmithorg/TestSmith/issues/1897). (Manual)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/JSFunctionExecution_spec.ts [hunk 1]
```diff
@@ -10,7 +10,8 @@ const jsEditor = ObjectsRegistry.JSEditor,
 
 let onPageLoadAndConfirmExecuteFunctionsLength: number,
   getJSObject: any,
-  functionsLength: number, jsObj: any;
+  functionsLength: number,
+  jsObj: any;
```
Comment (ApekshaBhosale) [inline app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/JSFunctionExecution_spec.ts:14]: can we add type here please?

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

**Review by ApekshaBhosale — DISMISSED** [review]

**Comment by ohansFavour** [comment]
> * [ ]  function name does not appear on debugger the first time error occurs
>   image
> * [ ]  The second time the same error occurs, function name appears but execution does not terminate and continues to run even in error state when explictly defined as async
> * [ ]  When code is in error state and is not async then correct error is displayed the second time
>   https://www.loom.com/share/9a08454394bc4f459aa92cdec0250134

@ramsaptami all of these issues are in release.

2 & 3 are related to https://github.com/appsmithorg/appsmith/issues/14863

**Comment by ramsaptami** [comment]
PR fixes only the issue related to showing errors on the debugger that were previously not present. This functionality has been tested on the DP

**Review by Rishabh-Rathod — DISMISSED** [review]
Looks good

File: app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/JSFunctionExecution_spec.ts [hunk 2]
```diff
@@ -170,8 +187,71 @@ describe("JS Function Execution", function() {
...
+    agHelper.GetNClick(jsEditor._runButton);
+    jsEditor.AssertParseError(false, true);
+    agHelper.GetNClick(locator._errorTab);
+    cy.contains(
+      "TypeError: Cannot read properties of undefined (reading 'name')",
+    ).should("not.exist");
+
+    // Re-introduce parse errors
+    jsEditor.EditJSObj(JS_OBJECT_WITH_PARSE_ERROR);
+    agHelper.GetNClick(jsEditor._runButton);
```
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/JSFunctionExecution_spec.ts:244]: @ohansFavour - is this comment addressed?

[the remark this answers is not in the document]
Comment (ohansFavour) [reply app/client/cypress/integration/Smoke_TestSuite/ServerSideTests/JsFunctionExecution/JSFunctionExecution_spec.ts:244]: Yes

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**